fix(headers): correct headers for endpoints

tthvo opened the pull request:
Fixes #123 
Related to #122 

Correct headers for endpoints since some do not return JSON-formatted response but still set `application/json` header.
Added tests for `/current` endpoint to native image while cleaning up tests.

I also enforce allowed VERB for each endpoints.

andrewazores reviewed (commented):
Looks good. Could you add some tests to check that response header and verify that you get an HTTP 405 response if you try to use the wrong verb on a given path?

maxcao13 reviewed (approved):
Looks good to me.

tthvo commented:
I added some check for headers and tests on invalid verbs and cleaned up some Mockito calls as mentioned in https://github.com/cryostatio/jfr-datasource/pull/115#issuecomment-1150467066. Any thoughts?
